### PR TITLE
change fetch to allow for local file selection

### PR DESCRIPTION
**Pull request by bigfoot1144**
you cannot currently use relative file paths with fetch

This shows up in the yolov8 example:

```
desktop>python3 tinygrad/examples/yolov8.py ./TestImages/Abyssinian_1.jpg
No variant given, so choosing 'n' as the default. Yolov8 has different variants, you can choose from ['n', 's', 'm', 'l', 'x']
running inference for YOLO version n
Traceback (most recent call last):
  File "tinygrad/examples/yolov8.py", line 405, in <module>
    image_location = [np.frombuffer(io.BytesIO(fetch(img_path)).read(), np.uint8)]
  File "/home/bigfoot/dev/tinyfork/tinygrad/extra/utils.py", line 22, in fetch
    download_file(url, fp, skip_if_exists=not getenv("NOCACHE"))
  File "/home/bigfoot/dev/tinyfork/tinygrad/extra/utils.py", line 39, in download_file
    r = requests.get(url, stream=True)
  File "/home/bigfoot/.local/lib/python3.8/site-packages/requests/api.py", line 73, in get
    return request("get", url, params=params, **kwargs)
  File "/home/bigfoot/.local/lib/python3.8/site-packages/requests/api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/bigfoot/.local/lib/python3.8/site-packages/requests/sessions.py", line 573, in request
    prep = self.prepare_request(req)
  File "/home/bigfoot/.local/lib/python3.8/site-packages/requests/sessions.py", line 484, in prepare_request
    p.prepare(
  File "/home/bigfoot/.local/lib/python3.8/site-packages/requests/models.py", line 368, in prepare
    self.prepare_url(url, params)
  File "/home/bigfoot/.local/lib/python3.8/site-packages/requests/models.py", line 439, in prepare_url
    raise MissingSchema(
requests.exceptions.MissingSchema: Invalid URL './TestImages/Abyssinian_1.jpg': No scheme supplied. Perhaps you meant https://./TestImages/Abyssinian_1.jpg?
```

also removed if statement in efficientnet to use fetch for local url

### CODE REVIEW

**Comment by geohot** [comment]
Needs test

**Comment by bigfoot1144** [comment]
Tests could potentially be done cleaner by splitting ../ and ./ into separate test classes. Not sure it's better or not.

Also I noticed fetch has 
```python
if url.startswith("/"):
```
the functionality of which is currently not tested.

**Comment by bigfoot1144** [comment]
Tested in both Windows and Ubuntu 20. Don't have a Mac machine to test.

File: test/extra/test_utils.py [hunk 1]
```diff
@@ -25,6 +25,27 @@ def test_fetch_img(self):
...
+  
+  #test ./
+  def test_fetch_relative_dotslash(self):
+    self.assertEqual(b'12345', fetch("./test_file.txt"))
+  
+  #test ../
+  def test_fetch_relative_dotdotslash(self):
+    os.chdir('test_file_path')
+    self.assertEqual(b'12345', fetch("../test_file.txt"))
+    os.chdir('../')
```
Comment (wozeparrot) [inline test/extra/test_utils.py:48]: Can you use the tempdir for these tests?

**Comment by bigfoot1144** [comment]
I've never used tempdir before. The implementation of the test feels a lot cleaner. 

Please tell me if I'm doing something stupid and I'll fix it.

**Comment by bigfoot1144** [comment]
Need to change to path for windows implementation...

**Comment by wozeparrot** [comment]
Thanks!

**Pull request merged**